seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java

```diff
@@ -0,0 +1,144 @@
+            Dataset<Row> ds = sources.get(0).getData(environment);
+            for (BaseSparkTransform tf : transforms) {
+                if (ds.head().size() > 0) {
+                    ds = SparkBatchExecution.transformProcess(environment, tf, ds);
+                    SparkBatchExecution.registerTransformTempView(tf, ds);
+                }
+            }
+
+            // if (ds.take(1).length > 0) {
+            for (BaseSparkBatchSink sink : sinks) {
+                SparkBatchExecution.sinkProcess(environment, sink, ds);
+            }
+            // }
+        }
+    }
+
+    @Override
+    public void setConfig(Config config) {
+        this.config = config;
+    }
+
+    @Override
+    public Config getConfig() {
+        return config;
+    }
+
+    @Override
+    public CheckResult checkConfig() {
+        return new CheckResult(true, "");
+    }
+
+    @Override
+    public void prepare(Void prepareEnv) {
+
+    }
+
+    public static void registerTempView(String tableName, Dataset<Row> ds) {
+        ds.createOrReplaceTempView(tableName);
+    }
+
+    public static void registerInputTempView(BaseSparkSource<Dataset<Row>> source, SparkEnvironment environment) {
+        Config conf = source.getConfig();
+        final boolean flag = conf.hasPath(SparkBatchExecution.RESULT_TABLE_NAME);
+
+        if (flag) {
+            String tableName = conf.getString(SparkBatchExecution.RESULT_TABLE_NAME);
+            registerTempView(tableName, source.getData(environment));
+        } else {
+            throw new ConfigRuntimeException(
+                    "Plugin[" + source.getClass().getName() + "] must be registered as dataset/table, please set \"result_table_name\" config");
+        }
+    }
+
+    public static Dataset<Row> transformProcess(SparkEnvironment environment, BaseSparkTransform transform, Dataset<Row> ds) {
+        Config config = transform.getConfig();
+        boolean flag = config.hasPath(SparkBatchExecution.SOURCE_TABLE_NAME);
+        Dataset<Row> fromDs;
+        if (flag) {
+            String sourceTableName = config.getString(SparkBatchExecution.SOURCE_TABLE_NAME);
+            fromDs = environment.getSparkSession().read().table(sourceTableName);
+
+        } else {
+            fromDs = ds;
+        }
+
+        return transform.process(fromDs, environment);
+    }
+
+
+    public static void registerTransformTempView(BaseSparkTransform plugin, Dataset<Row> ds) {
+        Config config = plugin.getConfig();
+        if (config.hasPath(SparkBatchExecution.RESULT_TABLE_NAME)) {
+            String tableName = config.getString(SparkBatchExecution.RESULT_TABLE_NAME);
+            registerTempView(tableName, ds);
+        }
+    }
+
+    public static void sinkProcess(SparkEnvironment environment, BaseSparkSink sink, Dataset<Row> ds) {
+        Config config = sink.getConfig();
+        boolean flag = config.hasPath(SparkBatchExecution.SOURCE_TABLE_NAME);
+        Dataset<Row> fromDs;
+        if (flag){
```

Comment:
better rename to a meaningfull variable.

Reply:
already solved